packages/backend/src/apps/formsg/triggers/new-submission/get-data-out-metadata.ts

```diff
@@ -103,7 +132,7 @@ async function getDataOutMetadata(
       fieldType: { isHidden: true },
       order: { isHidden: true },
       answerArray:
-        fieldData.answerArray && buildAnswerArrayMetadatum(fieldData),
+        isAnswerArrayValid(fieldData) && buildAnswerArrayMetadatum(fieldData),
```

Comment:
**Nit**: actually the earlier approach was better - the `&&` operator returns the LHS expression if it's falsy.

This means (for FormSG submissions that don't have an `answerArray` field) in the previous approach, `answerArray` prop in metadata would be undefined (which is expected). In your new approach, `answerArray` prop in metadata is `false` , which.... is not going to break things, but is weird.

Reply:
yea thks for catching it, i didnt realise it